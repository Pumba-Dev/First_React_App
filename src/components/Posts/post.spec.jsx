import { render, screen } from "@testing-library/react";
import {Posts} from ".";

const props = {
    posts: [
        {
            id: 1,
            title: "title 1",
            body: "body 1",
            cover: "/img1"
        },
        {
            id: 2,
            title: "title 2",
            body: "body 2",
            cover: "/img2"
        },
        {
            id: 3,
            title: "title 3",
            body: "body 3",
            cover: "/img3"
        }
    ]
}

describe('<Posts />', () => {
    it('should render posts correctly', () => {
        render(<Posts {...props} />);
        expect(screen.getAllByRole('img', {name: /title/i})).toHaveLength(props.posts.length);
        expect(screen.getAllByRole('heading', {name: /title/i})).toHaveLength(props.posts.length);
        expect(screen.getAllByText(/body/i)).toHaveLength(props.posts.length);
    });

    it('should match snapshot', () => {
        const {container} = render(<Posts {...props} />);
        expect(container.firstChild).toMatchSnapshot();
    });
});