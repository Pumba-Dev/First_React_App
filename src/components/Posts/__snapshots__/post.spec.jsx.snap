// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      src="/img1"
    />
    <div
      class="post-content"
    >
      <h1>
        title 1
      </h1>
      <p>
        body 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      src="/img2"
    />
    <div
      class="post-content"
    >
      <h1>
        title 2
      </h1>
      <p>
        body 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      src="/img3"
    />
    <div
      class="post-content"
    >
      <h1>
        title 3
      </h1>
      <p>
        body 3
      </p>
    </div>
  </div>
</div>
`;
